### PR TITLE
Hyrax::TikaFileCharacterizationService for file characterization

### PR DESCRIPTION
**Pull request by bmquinn**
- Adds `Hyrax::TikaFileCharacterizationService`
- Adds `Hyrax::MultiChecksum` model, which stores md5, sha1 and sha256 checksums for uploaded files
- Adds `file_nodes` attribute to `Hyrax::FileSetBehavior`
- Update template generator to add `Hyrax::TikaFileCharacterizationService` to `Valkyrie::FileCharacterizationService.services` in `valkyrie.rb`
- Adds `Hyrax::FileNode` and `Hyrax::Multichecksum` specs

### CODE REVIEW

File: app/models/concerns/hyrax/file_set_behavior.rb [hunk 1]
```diff
@@ -24,13 +24,15 @@ module FileSetBehavior
       self.human_readable_type = 'File'
 
       attribute :file_identifiers, Valkyrie::Types::Set
+      attribute :file_nodes, Valkyrie::Types::Set.member(FileNode.optional)
       attribute :member_ids, Valkyrie::Types::Array
+
+      delegate :width, :height, :mime_type, :size, to: :original_file, allow_nil: true
+      delegate :md5, :sha1, :sha256, to: :original_file_checksum, allow_nil: true
```
Comment (jcoyne) [inline app/models/concerns/hyrax/file_set_behavior.rb]: Where is `original_file_checksum` defined? I'm having trouble seeing it.

File: app/models/concerns/hyrax/file_set_behavior.rb [hunk 2]
```diff
@@ -24,13 +24,15 @@ module FileSetBehavior
       self.human_readable_type = 'File'
 
       attribute :file_identifiers, Valkyrie::Types::Set
+      attribute :file_nodes, Valkyrie::Types::Set.member(FileNode.optional)
```
Comment (jcoyne) [inline app/models/concerns/hyrax/file_set_behavior.rb]: What does this mean?  Is FileNode embeded in the FileSet?  Is that going to work in Fedora?

**Comment by jcoyne** [comment]
Needs rebase.

**Comment by mjgiarlo** [comment]
@jcoyne @kdid @jgondron one of y'all mind picking this up, if you have time this week?

**Comment by kdid** [comment]
@mjgiarlo - I can work on this.

**Comment by mjgiarlo** [comment]
@kdid :beers:

**Comment by kdid** [comment]
Closing. Opened new PR for this here: https://github.com/samvera/hyrax/pull/2337

**Pull request closed without merging**